Panic when unable to retrieve credentials from AWS STS

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x149d3de]

goroutine 770 [running]:
github.com/cortexproject/cortex/pkg/chunk/aws.dynamoDBRequestAdapter.Retryable(0xc42265a000, 0x1aa9ee2)
	/go/src/github.com/cortexproject/cortex/pkg/chunk/aws/dynamodb_storage_client.go:457 +0xe
github.com/cortexproject/cortex/pkg/chunk/aws.dynamoDBStorageClient.queryPage(0xc420478380, 0x4000000000000000, 0x0, 0x0, 0x0, 0x186a0, 0x3ff4cccccccccccd, 0x1b05684, 0x51, 0x1b0fe58, ...)
	/go/src/github.com/cortexproject/cortex/pkg/chunk/aws/dynamodb_storage_client.go:380 +0x38d
github.com/cortexproject/cortex/pkg/chunk/aws.dynamoDBStorageClient.query(0xc420478380, 0x4000000000000000, 0x0, 0x0, 0x0, 0x186a0, 0x3ff4cccccccccccd, 0x1b05684, 0x51, 0x1b0fe58, ...)
	/go/src/github.com/cortexproject/cortex/pkg/chunk/aws/dynamodb_storage_client.go:337 +0x472
github.com/cortexproject/cortex/pkg/chunk/aws.(dynamoDBStorageClient).(github.com/cortexproject/cortex/pkg/chunk/aws.query)-fm(0x1c655e0, 0xc4225eb050, 0xc420896300, 0x16, 0xc422532100, 0x3b, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cortexproject/cortex/pkg/chunk/aws/dynamodb_storage_client.go:285 +0xb4
github.com/cortexproject/cortex/pkg/chunk/util.DoParallelQueries.func1(0x1c655e0, 0xc4224e5890, 0xc4224a36e0, 0xc4224a3740, 0xc422656200, 0xc422648860)
	/go/src/github.com/cortexproject/cortex/pkg/chunk/util/util.go:42 +0x121
created by github.com/cortexproject/cortex/pkg/chunk/util.DoParallelQueries
	/go/src/github.com/cortexproject/cortex/pkg/chunk/util/util.go:34 +0xf9
```

--- a/pkg/chunk/aws/dynamodb_storage_client.go
+++ b/pkg/chunk/aws/dynamodb_storage_client.go
@@ -454,7 +454,7 @@ func (a dynamoDBRequestAdapter) HasNextPage() bool {
 }
 
 func (a dynamoDBRequestAdapter) Retryable() bool {
-	return *a.request.Retryable
+	return aws.BoolValue(a.request.Retryable)
 }
 
 type chunksPlusError struct {

